### ISSUE TITLE
Is it possible to add spaces to export and import declarations?

### ISSUE THREAD

**Issue by Maximaximum** [dsherret/ts-simple-ast#154]
Currently, when I use `SourceFile.addImport()`, I get such an output:
```
import {NetUser, User, SmthElse} from './models/User';
```

However, I'd like to **add a space after the opening bracket and before the closing one**, so that the file conforms to the style rules we use in our project:
```
import { NetUser, User, SmthElse } from './models/User';
```

The same with `addExport()`.

Is this achievable somehow?

*PS. I'm sorry I've created lots of issues in the last 2 days :) The project is really so awesome and I just want to use it at its full potential. Please, however, feel free to tell me if I'm misusing it and if the details of file formatting are out of scope of the project.*

**Comment by dsherret** [comment]
It's not easily possible at the moment. This is definitely something that I would like to add though... along with spaces between parenethesis (ex. `function func( param1, param2 )`). It'd definitely not out of scope.

Any ideas about what this manipulation setting should be called? Maybe we can look at tslint/eslint and call it `objectCurlySpacing: true` https://eslint.org/docs/rules/object-curly-spacing ?

For now, it's a bit of a hassel, but possible to do this by navigating the import declarations, getting all the points where a space should be inserted, then manually inserting whitepsace text via [`sourceFile.insertText`](https://dsherret.github.io/ts-simple-ast/manipulation/) (read warnings about using that on the linked page... that `insertText` method is really a last resort because it will blow away all previously navigated nodes within the source file)

No need to apologize! Please keep on creating issues because they're good to discuss and have in the issue tracker. I'm glad this project is of use to you :)

**Comment by Maximaximum** [comment]
Regarding the name of the option, I'm not sure. It seems like `ESLint`'s `objectCurlySpacing` does not check the import statements. However, `tslint` has a `whitespace` rule, which can be configured to check import declarations by providing the `check-module` key: https://palantir.github.io/tslint/rules/whitespace/

It feels like `ts-simple-ast` could potentially support all of the `tslint` rules that have fixes (after all, `tslint` is a solely typescript tool). Then it would be reasonable to conform to the `tslint` rule names.

If not, then it depends on whether this setting will be specific to only import/export declarations, or it will be general for all the whitespace around all the braces. In the first case, I think it should be called something like `import-whitespace` or `module-whitespace`. In the second case, `bracket-whitespace` could do.

**Comment by Maximaximum** [comment]
BTW, `SourceFile.formatText()` adds the whitespaces to the imports but, unfortunately, removes the empty lines 😆 https://github.com/dsherret/ts-simple-ast/issues/157

**Comment by Maximaximum** [comment]
https://github.com/vvakame/typescript-formatter supports this out of the box, so, in case you decide to implement `SourceFile.formatText()` via using typescript-formatter, this issue won't need any additional work to be done

**Comment by fbartho** [comment]
An alternate move might be to suggest emitting through something like prettier? That way the stylistic formatting is a distinct task than the AST manipulation.

**Comment by dsherret** [comment]
I'm going to change spaces to be the default because it's the default in the compiler (will be done in v10).

The ability to configure this will be added soon. It's much easier to implement it now because I've done a lot of internal refactoring to separate out this functionality. The setting for this is the `insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces` format code setting from the compiler.

By the way, instead of using `sourceFile.formatText()` it's possible to call `.formatText()` on any node to do select formats (ex. `importDeclaration.formatText()`).

Also, I don't believe I'll use prettier or typescript-formatter. I don't want to over-bloat the library (other points: prettier is opinionated and I don't want this library to be opinionated. typescript-formatter uses the same formatter used in the compiler API that's exposed in this library via `formatText()`). Plus, automatically formatting the text after every manipulation would really affect the performance... if someone really wants to format the text then it's recommended to do a batch of manipulations then call `formatText()`.

**Comment by dsherret** [comment]
This will be implemented in v10 when it's released.

Be default it will write surrounding spaces, but you can set the `insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces` manipulation setting to false to make it not add surrounding spaces.

**Comment by dsherret** [comment]
This is available now in 10.0.1.
